lib/std/os.zig

```diff
@@ -3327,3 +3327,37 @@ pub fn getrusage(who: i32) rusage {
         else => unreachable,
     }
 }
+
+pub const TermiosError = error {
+    NotATerminal,
+    InvalidAction,
+    ProcessOrphaned,
+} || UnexpectedError;
+
+pub fn tcgetattr(handle: fd_t) TermiosError!termios {
+    var term: termios = undefined;
+    switch (errno(system.tcgetattr(handle, &term))) {
```

Comment:
I think you also need the `EINTR` for this?

Reply:
No, [tcgetattr()](https://pubs.opengroup.org/onlinepubs/9699919799/functions/tcgetattr.html) doesn't return `EINTR`.

Reply:
The POSIX "shall fail if ..." listings aren't meant to be exhaustive. glibc and musl both implement `tcgetattr()` via [`ioctl()`](https://pubs.opengroup.org/onlinepubs/9699919799/functions/ioctl.html#tag_16_246_04). You may be right about `EINTR` being a non-issue, but I don't think its absence from the POSIX spec for `tcgetattr()` is sufficient to prove it.